js/server/modules/@arangodb/general-graph.js

```diff
@@ -31,69 +31,48 @@
 
 const internal = require('internal'); // OK: reloadAuth
 const ggc = require('@arangodb/general-graph-common');
-const GeneralGraph = internal.ArangoGeneralGraphModule;
-
-// This is supposed to be a class
-const ArangoGraph = internal.ArangoGraph;
-
-// inherited graph class
-let CommonGraph = ggc.__GraphClass;
-
-// new c++ based
-CommonGraph.prototype.__updateDefinitions = function (edgeDefs, orphans) {
-  this.__edgeDefinitions = edgeDefs;
-  this.__orphanCollections = orphans;
-};
-
-CommonGraph.prototype._deleteEdgeDefinition = function (edgeDefinition, dropCollection = false) {
-  let result = ArangoGraph._deleteEdgeDefinition(this.__name, edgeDefinition, dropCollection);
-  this.__updateDefinitions(result.graph.edgeDefinitions, result.graph.orphanCollections);
-};
-
-CommonGraph.prototype._extendEdgeDefinitions = function (edgeDefinitions, options = {}) {
-  let result = ArangoGraph._extendEdgeDefinitions(this.__name, edgeDefinitions, options);
-  this.__updateDefinitions(result.graph.edgeDefinitions, result.graph.orphanCollections);
+const GeneralGraphModule = internal.ArangoGeneralGraphModule;
+
+const {GeneralGraph, SmartGraph, EnterpriseGraph, SatelliteGraph} = internal.isEnterprise() ?
+  require('@arangodb/ee-graph-classes') : require('@arangodb/graph-classes');
+
+const graphToClass = graph => {
```

Comment:
Same method as in : `js/client/modules/@arangodb/general-graph.js`. Could be exported / unified - but very minor.

Reply:
I think it cannot, because it uses the above require.
Where the required files are different in client and server